sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -365,10 +365,17 @@ cdef int free_tree(Tree* tree) nogil:
     for i in range(3):
         cnt[i] = 0
     free_recursive(tree, tree.root_node, cnt)
+    if not tree.root_node.is_leaf:
+        free(tree.root_node.children)
+    free(tree.root_node.width)
+    free(tree.root_node.left_edge)
+    free(tree.root_node.center)
+    free(tree.root_node.barycenter)
+    free(tree.root_node.leaf_point_position)
     free(tree.root_node)
```

Comment:
I have just wondering if it would not be cleaner to re-arrange the way things are done in the `free_recursive` loop so that we don't need to duplicate this sequence of free for the root in the caller.